### PR TITLE
BFF add health endpoint

### PR DESCRIPTION
**Pull request by djorgensendk**

### CODE REVIEW

**Comment by djorgensendk** [comment]
@sla-bot

**Comment by djorgensendk** [comment]
@sla-bot check

**Comment by djorgensendk** [comment]
@cla-bot check

File: apps/dh/api-dh/source/DataHub.WebApi/Startup.cs [hunk 1]
```diff
@@ -141,11 +146,38 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment environment)
...
+                .AddLiveCheck()
+                .AddServiceHealthCheck("charges", chargesLiveHealthUrl)
+                .AddServiceHealthCheck("marketParticipant", marketParticipantLiveHealthUrl)
+                .AddServiceHealthCheck("messageArchive", messageArchiveLiveHealthUrl)
+                .AddServiceHealthCheck("wholesale", wholesaleLiveHealthUrl);
+        }
+
+        private ApiClientSettings AddDomainClients(IServiceCollection services)
         {
             var apiClientSettings = Configuration.GetSection("ApiClientSettings").Get<ApiClientSettings>();
```
Comment (dstenroejl) [inline apps/dh/api-dh/source/DataHub.WebApi/Startup.cs:182]: Move this to `ConfigureServices` and add a parameter to `AddDomainClients`. Then consider if `AddDomainClients` and `SetupHealthEndpoints` can be called just after one another, so we easily see their relationship. But beware, sometimes the order of registration is important.

**Review by dstenroejl — COMMENTED** [review]
Added a few suggestions.

File: apps/dh/api-dh/source/DataHub.WebApi.Tests/TestHelpers/AuthorizationConfigurationData.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections.Generic;
+using Energinet.DataHub.Core.FunctionApp.TestCommon.Configuration.B2C;
+
+namespace Energinet.DataHub.WebApi.Tests.TestHelpers
```
[the remark this answers is not in the document]
Comment (djorgensendk) [reply apps/dh/api-dh/source/DataHub.WebApi.Tests/TestHelpers/AuthorizationConfigurationData.cs]: The class has been removed, and content moved into BffWebApiFixture

File: apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/Controllers/ChargeLinksControllerTests.cs [hunk 3]
```diff
@@ -19,59 +19,83 @@
...
+        IClassFixture<WebApiFactory>,
+        IAsyncLifetime
     {
-        private Fixture Fixture { get; }
+        private Fixture DtoFixture { get; }
 
         private Mock<IChargeLinksClient> ApiClientMock { get; }
 
-        private HttpClient HttpClient { get; }
+        private readonly HttpClient _client;
```
[the remark this answers is not in the document]
Comment (djorgensendk) [reply apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/Controllers/ChargeLinksControllerTests.cs]: changed to a property

File: apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/Controllers/MeteringPointControllerTests.cs [hunk 4]
```diff
@@ -18,57 +18,80 @@
...
+        IClassFixture<BffWebApiFixture>,
+        IClassFixture<WebApiFactory>,
+        IAsyncLifetime
     {
-        public MeteringPointControllerTests(WebApplicationFactory<Startup> factory)
+        private Fixture DtoFixture { get; }
+
+        private Mock<IMeteringPointClient> ApiClientMock { get; }
+
+        private readonly HttpClient _client;
```
[the remark this answers is not in the document]
Comment (djorgensendk) [reply apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/Controllers/MeteringPointControllerTests.cs]: changed to a readonly property

File: apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/EndpointTests.cs [hunk 5]
```diff
@@ -28,11 +29,20 @@ namespace Energinet.DataHub.WebApi.Tests.Integration
...
     {
-        public class GetOpenApiDocumentation : WebHostTestBase
+        public class GetOpenApiDocumentation :
+                    WebApiTestBase<BffWebApiFixture>,
+                    IClassFixture<BffWebApiFixture>,
+                    IClassFixture<WebApiFactory>
         {
-            public GetOpenApiDocumentation(WebApplicationFactory<WebApi.Startup> factory)
-                : base(factory)
+            private readonly HttpClient _client;
```
[the remark this answers is not in the document]
Comment (djorgensendk) [reply apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/EndpointTests.cs]: changed to a property

File: apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/EndpointTests.cs [hunk 6]
```diff
@@ -53,11 +63,20 @@ public async Task When_StandardRequest_Then_ResponseIsOKAndContainsJsonAndOpenAP
...
 
-            public class GetSwaggerUI : WebHostTestBase
+            public class GetSwaggerUI :
+                    WebApiTestBase<BffWebApiFixture>,
+                    IClassFixture<BffWebApiFixture>,
+                    IClassFixture<WebApiFactory>
             {
-                public GetSwaggerUI(WebApplicationFactory<WebApi.Startup> factory)
-                    : base(factory)
+                private readonly HttpClient _client;
```
[the remark this answers is not in the document]
Comment (djorgensendk) [reply apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/EndpointTests.cs]: changed to a property

File: apps/dh/api-dh/source/DataHub.WebApi/SwaggerHostFactory.cs [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+
+namespace Energinet.DataHub.WebApi
+{
+    public class SwaggerHostFactory
```
[the remark this answers is not in the document]
Comment (djorgensendk) [reply apps/dh/api-dh/source/DataHub.WebApi/SwaggerHostFactory.cs:24]: documentation added to this class and SwaggerStartup

File: apps/dh/api-dh/source/DataHub.WebApi/Startup.cs (hunk 1 above)
Comment (djorgensendk) [reply apps/dh/api-dh/source/DataHub.WebApi/Startup.cs]: done

File: apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/HealthCheckTests.cs [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+using Energinet.DataHub.Core.App.Common.Diagnostics.HealthChecks;
+using Energinet.DataHub.WebApi.Tests.Fixtures;
+using FluentAssertions;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Energinet.DataHub.WebApi.Tests.Integration
+{
+    /*
```
[the remark this answers is not in the document]
Comment (djorgensendk) [reply apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/HealthCheckTests.cs]: WireMock is used to act as the server responding to the requests for all of the ready checks

File: apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/HealthCheckTests.cs [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+            // Assert
+            actualResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+
+            var actualContent = await actualResponse.Content.ReadAsStringAsync();
+            actualContent.Should().Be(Enum.GetName(typeof(HealthStatus), HealthStatus.Healthy));
+        }
+
+        [Fact]
+        public async Task When_RequestReadinessStatus_Then_ResponseIsOkAndHealthy()
```
Comment (dstenroejl) [inline apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/HealthCheckTests.cs:83]: I believe we should have a test that shows Unhealthy as well. We could configure 3 endpoints to return OK and then just one to fail.

Comment (dstenroejl) [inline apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/HealthCheckTests.cs:83]: I believe we should have a test that shows Unhealthy as well. We could configure 3 endpoints to return OK and then just one to fail.

**Review by dstenroejl — COMMENTED** [review]
Just one more test showing a failing HC. :)

File: apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/HealthCheckTests.cs (hunk 9 above)
Comment (djorgensendk) [reply apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/HealthCheckTests.cs:83]: added - test named If_ADependentServiceIsUnavailable_When_RequestReadinessStatus_Then_ResponseIsServiceUnavailableAndUnhealthy

**Review by dstenroejl — APPROVED** [review]

**Pull request merged**